Editor/CoreAPI/StateManager.cs

```diff
@@ -206,6 +206,22 @@ public string DeploymentBuildFolderPath
 
         #endregion
 
+        public bool JustLaunchedServer
+        {
+            get => CoreApi.GetSetting(SettingsKeys.JustLaunchedServer).Value == "true";
+            set
+            {
+                if (value)
+                {
+                    CoreApi.PutSetting(SettingsKeys.JustLaunchedServer, "true");
+                }
+                else
+                {
+                    CoreApi.ClearSetting(SettingsKeys.JustLaunchedServer);
+                }
+            }
+        }
+
```

Comment:
This seems like extremely tailor made code, and we should avoid adding settings that are not necessary. 

Ideally in my mind the `GetComputeAuthToken` is something that the customer should get before starting up the server.  The text in the UX states `The auth token is generated when you launch the game`. So we should be displaying the parameter that was used to launch the server. There are 2 options here

1. Pass in the authToken to the server somehow (i'm not sure this is feasible)
2. Write the authToken in the settings instead, and have a callback that updates the UI when the authToken is updated in the stateManager. That way if we decide to refresh the authToken in the future, there's an inbuilt mechanism that refreshes it on the UX as well.

Reply:
1. Because the server is handled by the sample game there is no nice way that wouldn't cause issues with waiting for variables to be set before running the server. They are 2 seperate entities so this was the best solution we could come up with. 
2. The problem is that when we were initially discussing the generation of authtoken in the client via the plugin, we were told by James that for security reasons we cannot store that variable anywhere. 

This is why we had to come up with a odd way of solving this requirement for the UX. As being generated within the same 15 mins as the game launches, will give us back the same token to display. And when we launch the game it will refresh the UI so this would trigger us getting the authtoken on starting the game.